### ISSUE TITLE
When attempting to run drush ": No such file or directory"

### ISSUE THREAD

**Issue by jmk1ng** [drush-ops/drush#418]
I've been bashing my head against the wall all day trying to figure out why Drush has suddenly stopped working.

I've re-installed via composer, tried different versions, always the same error.

If I type `which drush` I get `/home/user/.composer/vendor/bin/drush` which is correct.

If I type `sh -vx drush` in an attempt to debug what's happening with the script, I get the following output:

```

#!/usr/bin/env sh
#
# This script is a simple wrapper that will run Drush with the most appropriate
# php executable it can find.
#
# Solaris users: Add /usr/xpg4/bin to the head of your PATH
#

+ 
: not found/.composer/vendor/bin/drush: 8: /home/user/.composer/vendor/bin/drush: 
# Get the absolute path of this executable
SELF_DIRNAME="`dirname -- "$0"`"
+ dirname -- /home/user/.composer/vendor/bin/drush
+ SELF_DIRNAME=/home/user/.composer/vendor/bin
SELF_PATH="`cd -P -- "$SELF_DIRNAME" && pwd -P`/`basename -- "$0"`"
+ cd -P -- /home/user/.composer/vendor/bin
/home/user/.composer/vendor/bin/drush: 1: cd: can't cd to /home/user/.composer/vendor/bin
+ basename -- /home/user/.composer/vendor/bin/drush
+ SELF_PATH=/drush

+ 
: not found/.composer/vendor/bin/drush: 12: /home/user/.composer/vendor/bin/drush: 
# Decide if we are running a Unix shell on Windows
if `which uname > /dev/null 2>&1`; then
  case "`uname -a`" in
/home/user/.composer/vendor/bin/drush: 15: /home/user/.composer/vendor/bin/drush: Syntax error: word unexpected (expecting "in")
```

wut

Ok, maybe something wonky is returned with `uname -a`? The response from my server is:

`Linux servername 3.2.0-58-generic #88-Ubuntu SMP Tue Dec 3 17:37:58 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux`

There are no directory permission issues... everything is owned by the user. I'm at a complete loss. I've tried drush 5, 6, and 7 - same sort of error in each.

The server is running Ubuntu 12.04.2 LTS

**Comment by ratajczak** [comment]
I have the same issue, it's caused by dos line endings.
I've fixed it by editing this file https://github.com/drush-ops/drush/blob/master/drush in vim and executing ":set ff=unix"

**Comment by andyg5000** [comment]
My issue was caused by the .bashrc not being processed by non interactive shells http://stackoverflow.com/a/941995/980921

**Comment by damienmckenna** [comment]
FYI, drush was working for me when I manually ran it via ssh but failed with this error when I tried doing remote commands. It turned out the problem was that drush was installed via to a custom directory and then I had edited `~/.bash_profile~ to make it run, which allowed it to run when I manually ssh'd to the server. However, this file isn't executed when you connect via ssh command execution, which Drush uses to connect remotely, so I had to also add the path export line to `~/.bashrc`. It now works correctly.
